planner/core/logical_plans_test.go

```diff
@@ -2064,6 +2064,36 @@ func TestUpdateEQCond(t *testing.T) {
 	}
 }
 
+func TestApplyEliminate(t *testing.T) {
+	tests := []struct {
+		sql  string
+		best string
+	}{
+		{
+			sql:  "select count(*) from (select a,(select t2.b from t t2,t t3 where t2.a=t3.a and t1.a=t2.a limit 1) t from t t1) t",
+			best: "DataScan(t1)->Aggr(count(1))->Projection",
```

Comment:
The test case is not enough, can you add 2 other cases:
1. select count(a) from (select a, () ....) t
2. select count(b) from (select a, (select t2.b....) as b  ....) t